melodies_monet/driver.py

```diff
@@ -860,13 +984,33 @@ def plotting(self):
                         # Adjust the modvar as done in pairing script, if the species name in obs and model are the same.
                         if obsvar == modvar:
                             modvar = modvar + '_new'
-
-                        # convert to dataframe
-                        pairdf_all = p.obj.to_dataframe(dim_order=["time", "x"])
-
-                        # Select only the analysis time window.
-                        pairdf_all = pairdf_all.loc[self.start_time : self.end_time]
-
+                            
+                        # for pt_sfc data, convert to pandas dataframe, format, and trim
+                        if obs_type == 'pt_sfc':
+                            # convert to dataframe
+                            pairdf_all = p.obj.to_dataframe(dim_order=["time", "x"])
+                            # Select only the analysis time window.
+                            pairdf_all = pairdf_all.loc[self.start_time : self.end_time]
+                        
+                        # keep data in xarray, fix formatting, and trim
+                        elif obs_type in ["sat_swath_sfc", "sat_swath_clm", 
+                                                                        "sat_grid_sfc", "sat_grid_clm", 
+                                                                        "sat_swath_prof"]:
+                             # convert index to time; setup for sat_swath_clm
+                            
+                            if 'time' not in p.obj.dims and obs_type == 'sat_swath_clm':
+                                
+                                pairdf_all = p.obj.swap_dims({'x':'time'})
+                            # squash lat/lon dimensions into single dimension
+                            elif obs_type == 'sat_grid_clm':
+                                pairdf_all = p.obj.stack(ll=['x','y'])
+                                pairdf_all = pairdf_all.rename_dims({'ll':'y'})
+                            else:
+                                pairdf_all = p.obj
```

Comment:
Not necessary to work this out now but if keeping paired data as xarray for satellite obs, "pairdf" and more generally anything with "df" becomes a misleading variable name.

Reply:
Should this be paried_dataframe instead?

Reply:
I guess moreso I meant that pairdf would be interpreted to mean this variable is a pandas dataframe, but in the satellite processing this is actually an xarray dataset. It might be better to refer to it more generally as pair_data or something similar. Again. not a critical thing to address but for future development it would be more clear to try not to have variable names including "df" when when the variable is not a pandas dataframe.

Reply:
We will defer this to a future issue, as it looks like there may be some additional reorganization needed.
We have created a new issue https://github.com/NOAA-CSL/MELODIES-MONET/issues/193.